### PR TITLE
Refactor Gdrive to GCS operator

### PR DESCRIPTION
**Pull request by Scuall1992**
---

@turbaszek Should I create an issue for this?

**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 1]
```diff
@@ -74,8 +73,8 @@ class GoogleDriveToGCSOperator(BaseOperator):
     def __init__(
         self,
         *,
-        destination_bucket: str,
-        destination_object: str,
+        bucket_name: str,
+        object_name: str,
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]: This is a breaking change. While I agree that you suggestion is reasonable and I support it we will need to keep old arguments to preserve backward compatibly and we need to add deprecation warning.

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 2]
```diff
@@ -129,4 +104,10 @@ def execute(self, context):
...
             impersonation_chain=self.impersonation_chain,
         )
-        self._upload_data(gdrive_hook=gdrive_hook, gcs_hook=gcs_hook)
+        file_metadata = gdrive_hook.get_file_id(
+            folder_id=self.folder_id, file_name=self.file_name, drive_id=self.drive_id
+        )
+        with gcs_hook.provide_file_and_upload(
+            bucket_name=self.bucket_name, object_name=self.object_name
+        ) as file:
+            gdrive_hook.download_file(file_id=file_metadata["id"], file_handle=file)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py:126]: Should we also adjust tests?

Comment (Scuall1992) [reply airflow/providers/google/cloud/transfers/gdrive_to_gcs.py:126]: Yes, I'll do it today

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py (hunk 1 above)
Comment (Scuall1992) [reply airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]: Okay

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 3]
```diff
@@ -85,38 +92,14 @@ def __init__(
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.destination_bucket = destination_bucket
-        self.destination_object = destination_object
+        self.bucket_name = destination_bucket or bucket_name
+        self.object_name = destination_object or object_name
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
        self.bucket_name = destination_bucket or bucket_name
        if destination_bucket:
            warnings.warn("`destination_bucket ` is deprecated please use `bucket_name`", DeprecationWarning)
            bucket_name = destination_bucket
        self.bucket_name = bucket_name
```
In this way users will get DeprecationWarning in runtime, WDYT? Probably we should do the same for `object_name`.

Comment (Scuall1992) [reply airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]: I agree. Good suggestion

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
+from logging import warning
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
import warnings
```

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 5]
```diff
@@ -93,7 +94,11 @@ def __init__(
     ) -> None:
         super().__init__(**kwargs)
         self.bucket_name = destination_bucket or bucket_name
+        if destination_bucket:
+            warning.warn("`destination_bucket ` is deprecated please use `bucket_name`", DeprecationWarning)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
            warnings.warn("`destination_bucket ` is deprecated please use `bucket_name`", DeprecationWarning)
```

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 6]
```diff
@@ -93,7 +94,11 @@ def __init__(
...
         self.object_name = destination_object or object_name
+        if destination_object:
+            warning.warn("`destination_object ` is deprecated please use `object_name`", DeprecationWarning)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
            warnings.warn("`destination_object ` is deprecated please use `object_name`", DeprecationWarning)
```

File: tests/providers/google/cloud/transfers/test_gdrive_to_gcs.py [hunk 7]
```diff
@@ -85,13 +46,14 @@ def test_execute(self, mock_upload_data, mock_gdrive_hook, mock_gcs_hook):
...
+        mock_gdrive_hook.return_value.get_file_id.assert_called_once_with(
+            folder_id=FOLDER_ID, file_name=FILE_NAME, drive_id=DRIVE_ID
         )
-        mock_gcs_hook.assert_called_once_with(
-            gcp_conn_id=GCP_CONN_ID,
-            delegate_to=None,
-            impersonation_chain=IMPERSONATION_CHAIN,
+
+        mock_gdrive_hook.return_value.download_file.assert_called_once_with(
+            file_id=mock.ANY, file_handle=mock.ANY
```
Comment (turbaszek) [inline tests/providers/google/cloud/transfers/test_gdrive_to_gcs.py]: I think we should test that the `file_id` is passed properly. So I would suggest adding this before `op.execute`:
```
meta = {"id": "123xyz"}
mock_gdrive_hook.return_value.get_file_id.return_value = meta
```
and then we can do
```suggestion
        mock_gdrive_hook.return_value.download_file.assert_called_once_with(
            file_id=meta["id"], file_handle=mock.ANY
```

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 8]
```diff
@@ -85,38 +93,18 @@ def __init__(
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.destination_bucket = destination_bucket
-        self.destination_object = destination_object
+        self.bucket_name = destination_bucket or bucket_name
+        if destination_bucket:
+            warnings.warn("`destination_bucket ` is deprecated please use `bucket_name`", DeprecationWarning)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
            warnings.warn("`destination_bucket` is deprecated please use `bucket_name`", DeprecationWarning)
```

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py [hunk 9]
```diff
@@ -85,38 +93,18 @@ def __init__(
...
+        self.object_name = destination_object or object_name
+        if destination_object:
+            warnings.warn("`destination_object ` is deprecated please use `object_name`", DeprecationWarning)
```
Comment (turbaszek) [inline airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]:
```suggestion
            warnings.warn("`destination_object` is deprecated please use `object_name`", DeprecationWarning)
```

**Review by turbaszek — APPROVED** [review]
Only a small issue, otherwise this looks really good to me! 🚀

File: airflow/providers/google/cloud/transfers/gdrive_to_gcs.py (hunk 9 above)
Comment (Scuall1992) [reply airflow/providers/google/cloud/transfers/gdrive_to_gcs.py]: I can do it in an 3-4 hours

**Comment by Scuall1992** [comment]
> Only a small issue, otherwise this looks really good to me! 🚀

Thank you for your help. Do you have any other issue for me?

**Comment by Scuall1992** [comment]
@turbaszek Is there anything that I need to do in this issue?

**Review by turbaszek — APPROVED** [review]

**Comment by Scuall1992** [comment]
How can I restart checks?

**Comment by turbaszek** [comment]
> How can I restart checks?

I restarted them. In general `git commit --amend --no-edit` and `git push --force-with-lease` will trigger them once more

**Comment by Scuall1992** [comment]
I can't figure out what is go wrong

**Comment by turbaszek** [comment]
@Scuall1992 can you try rebasing onto current master? It's possible that we have some problems with CI...

**Comment by Scuall1992** [comment]
@turbaszek  I did it. Can you check please?

**Comment by turbaszek** [comment]
@Scuall1992 great job! 👏

**Comment by Scuall1992** [comment]
@turbaszek thank you. Do you have any ideas what I can do next?

**Comment by turbaszek** [comment]
@Scuall1992 there's a loot of `good-first-issues` that need some love:
https://github.com/apache/airflow/contribute

**Pull request merged**